### PR TITLE
Initial implementation of findDevices()

### PR DESCRIPTION
**Pull request by lmorchard**
Here's a thing that finds FxOS devices. Fancy a review, @tofumatt or @sole?

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+// See https://github.com/jshint/jshint/issues/1747 for context
+/* global -Promise */
+var Promise = require('es6-promise').Promise;
+var adb = require('adbkit');
+
+module.exports = findDevices;
+
+// Stolen from the ADB Helper add-on
+// https://github.com/mozilla/adbhelper/blob/36559184f122f410a4e571f6217e6ed8ca4fa33e/scanner.js#L114
+var B2G_TEST_COMMAND = 'test -f /system/b2g/b2g; echo $?';
```
Comment (fwenzel) [inline index.js:12]: that is... interesting :)

File: tests/unit/test.index.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+      .returns(new Promise(function(resolve, reject) {
+        resolve([
+          { id: fxosId, type: 'device' },
+          { id: androidId, type: 'device' }
+        ]);
+      }));
+
+    var mockShell = function(deviceId, command) {
+      return new Promise(function(resolve, reject) {
+        // Note: '0' means FxOS *was* detected.
```
Comment (fwenzel) [inline tests/unit/test.index.js:26]: good comments. Counterintuitive indeed.

**Comment by fwenzel** [comment]
lgtm, r+

File: LICENSE [hunk 3]
```diff
@@ -0,0 +1,13 @@
+Copyright 2014 Mozilla
```
Comment (tofumatt) [inline LICENSE]: It's 2015, dawg.

File: tests/unit/test.index.js [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+    findDevices().catch(function(err) {
+      test.ifError(err);
+      test.done();
+    }).then(function(results) {
+
+      // Ensure all the mocks were called, and with the expected parameters
+      test.ok(mocked.assert());
+
+      var foundFirefoxOS = false;
+      results.forEach(function(result) {
```
Comment (fwenzel) [inline tests/unit/test.index.js:67]: this test case seems a tiny little bit roundabout. but probably not too confusing to warrant rewriting.

File: tests/unit/test.index.js [hunk 5]
```diff
@@ -0,0 +1,81 @@
+'use strict';
+
+/* global -Promise */
+var Promise = require('es6-promise').Promise;
+var mockery = require('mockery');
+var nodemock = require('nodemock');
+
+module.exports = {
+
+  'findDevices() should work': function(test) {
```
Comment (tofumatt) [inline tests/unit/test.index.js]: This could be a bit more explain-y... what does "working" mean? :smile:

**Comment by tofumatt** [comment]
Mostly looks good to me. Really just see my comment about `'findDevices() should work'`, because the test is a bit confusing to follow so explaining what it's testing for might help folks new to the codebase.

File: tests/unit/test.index.js (hunk 5 above)
Comment (lmorchard) [inline tests/unit/test.index.js]: Oh, derp. That's from way earlier when the test body was just `test.done()` :dizzy_face:

File: LICENSE (hunk 3 above)
Comment (lmorchard) [inline LICENSE]: Copypasta bites me again!

**Comment by lmorchard** [comment]
Made some tweaks in response to comments!

**Comment by tofumatt** [comment]
Sweet :+1:

**Pull request merged**